go/blob/azure.go

```diff
@@ -48,9 +54,9 @@ type AzureBlobBucket struct {
 // NewAzureBlobBucket creates an Azure Blob object storage bucket. At least one
 // AzureConfigOption must be provided to specify authentication; either storage
 // account key or SAS token.
-func NewAzureBlobBucket(ctx context.Context, container string, accountName string, opts []AzureConfigOption) (*AzureBlobBucket, error) {
-	cfg := azureBlobStoreConfig{endpoint: defaultAzureEndpoint}
-	for _, opt := range opts {
+func NewAzureBlobBucket(ctx context.Context, container string, accountName string, authOpt AzureConfigOption, otherOpts ...AzureConfigOption) (*AzureBlobBucket, error) {
```

Comment:
I reworked the function signature here and for `NewGCSBucket` to explicitly require an option for authentication, since that is something that is needed anyway, and most callers will only provide that one option.